diff -urw a/mkspecs/features/java.prf b/mkspecs/features/java.prf
--- a/mkspecs/features/java.prf	2025-06-05 16:07:26.000000000 +0200
+++ b/mkspecs/features/java.prf	2026-05-31 15:27:26.603607991 +0200
@@ -23,9 +23,11 @@
 
 # Check for user defined javac build version
 javac_target_version = $$ANDROID_JAVAC_TARGET_VERSION
+isEmpty(javac_target_version): javac_target_version = 8
 isEmpty(javac_target_version): javac_target_version = 7
 
 javac_source_version = $$ANDROID_JAVAC_SOURCE_VERSION
+isEmpty(javac_source_version): javac_source_version = 8
 isEmpty(javac_source_version): javac_source_version = 7
 
 javac.commands = javac -source $$javac_source_version -target $$javac_target_version -Xlint:unchecked -bootclasspath $$ANDROID_JAR_FILE -cp $$shell_quote($$system_path($$join(JAVACLASSPATH, $$DIRLIST_SEPARATOR))) -d $$shell_quote($$CLASS_DIR) ${QMAKE_FILE_IN}
diff -urw a/src/network/ssl/ssl.pri b/src/network/ssl/ssl.pri
--- a/src/network/ssl/ssl.pri	2025-06-05 16:07:26.000000000 +0200
+++ b/src/network/ssl/ssl.pri	2026-05-31 15:28:02.475863735 +0200
@@ -116,6 +116,7 @@
         #   - configure: -openssl -openssl-linked -I <OPENSSL_DIR>\include -L <OPENSSL_DIR>\lib\VC\static OPENSSL_LIBS="-lUser32 -lAdvapi32 -lGdi32" OPENSSL_LIBS_DEBUG="-lssleay32MDd -llibeay32MDd" OPENSSL_LIBS_RELEASE="-lssleay32MD -llibeay32MD"
 
         qtConfig(openssl-linked): {
+            false: \
             android {
                 build_pass|single_android_abi: LIBS_PRIVATE += -lssl_$${QT_ARCH} -lcrypto_$${QT_ARCH}
             } else: QMAKE_USE_FOR_PRIVATE += openssl
